src/nanoemoji/util.py

```diff
@@ -56,3 +56,34 @@ def fs_root() -> Path:
 def rel(from_path: Path, to_path: Path) -> Path:
     # relative_to(A,B) doesn't like it if B doesn't start with A
     return Path(os.path.relpath(str(to_path.resolve()), str(from_path.resolve())))
+
+
+def build_n_ary_tree(leaves, n):
+    """Build N-ary tree from sequence of leaf nodes.
+
+    Return a list of lists where each non-leaf node is a list containing
+    max n nodes.
```

Comment:
your asserts preclude a tree of no nodes being an empty list ... why would that be wrong?